cavityFaultDisplay.py

```diff
@@ -0,0 +1,57 @@
+from PyQt5.QtWidgets import QHBoxLayout, QLabel, QSizePolicy, QVBoxLayout
+from functools import partial
+from pydm import Display
+from typing import List
+
+from displayCavity import DISPLAY_LINAC_OBJECTS
+from Fault import Fault, PvInvalid
+
+
+class CavityFaultDisplay(Display):
+    def __init__(self, parent=None, args=None, macros=None):
+        super().__init__(parent=parent, args=args,
+                         ui_filename="frontend/cavityfaultdisplay.ui",
+                         macros=macros)
+
+        linacIdx = int(macros["linac"][1])
+        cryomoduleName = macros["cryoNum"]
+        cavityNumber = macros["cavityNumber"]
+
+        cavityObject = DISPLAY_LINAC_OBJECTS[linacIdx].cryomodules[cryomoduleName].cavities[cavityNumber]
+
+        faults: List[Fault] = cavityObject.faults
+        verticalLayout: QVBoxLayout = self.ui.cavityfaultslayout
+
+        for fault in faults:
+            horizontalLayout = QHBoxLayout()
+            statusLabel = QLabel()
+            statusLabel.setStyleSheet("font-weight: bold")
+            statusLabel.setSizePolicy(QSizePolicy.MinimumExpanding,
+                                      QSizePolicy.MinimumExpanding)
+
+            codeLabel = QLabel()
+            codeLabel.setText(fault.tlc)
+            codeLabel.setSizePolicy(QSizePolicy.Maximum, QSizePolicy.Maximum)
+
+            nameLabel = QLabel()
+            nameLabel.setText(fault.name)
+            nameLabel.setSizePolicy(QSizePolicy.Maximum, QSizePolicy.Maximum)
+
+            horizontalLayout.addWidget(codeLabel)
+            horizontalLayout.addWidget(nameLabel)
+            horizontalLayout.addWidget(statusLabel)
+
+            verticalLayout.addLayout(horizontalLayout)
+            self.statusLabelCallback(statusLabel, fault)
+
+            fault.pv.add_callback(partial(self.statusLabelCallback, statusLabel, fault))
```

Comment:
What's the difference between fault.pv and fault in this line?

Reply:
fault is a fault object, as seen in Fault.py. Fault objects have a parameter called pv that stores a PV object, and that's what this is referencing